packages/dashboard-message-bus-client/lib/client.ts

```diff
@@ -0,0 +1,223 @@
+import { DashboardMessageBusClientOptions } from "./types";
+import { createMessage, Message } from "@truffle/dashboard-message-bus";
+
+import { DashboardMessageBusConnection } from "./connection";
+interface SendArgs<M extends Message> {
+  type: M["type"];
+  payload: M["payload"];
+  bestEffort?: boolean;
+}
+
+export class MessageBusConnectionError extends Error {
+  cause: Error | undefined;
+
+  constructor({ message, cause }: { message: string; cause?: Error }) {
+    super(message);
+    this.cause = cause;
+  }
+  toString(): string {
+    return this.cause
+      ? `${super.toString()}\ncaused by: ${this.cause.toString()}`
+      : super.toString();
+  }
+}
+
+export class DashboardMessageBusClient {
```

Comment:
Let's get this at the top of the file, with only the Options type defined above it :)

Reply:
haha, yep - looks like the changes I pushed last night covered that off already :-)